Test suite failing if run as root (inside docker container)

I'm trying to build a conda package out of this project. The builds work fine on OSX, for both python 2.7 and 3.6, but I'm stuck with the test suite on Linux (tested via a docker container based on continuumio/conda-concourse-ci:latest). Errors repeat for any of these two versions of Python. The following error occurs:

```text
+ nosetests test/test_account_info.py test/test_b2http.py test/test_base.py test/test_bucket.py test/test_console_tool.py test/test_download_dest.py test/test_exception.py test/test_parse_args.py test/test_policy.py test/test_progress.py test/test_sync.py test/test_sync_report.py test/test_utils.py
...............................................................................
WARNING: Unable to print unicode.  Encoding for stdout is: '<MagicMock name='mock.encoding' id='140584093796280'>'
Trying to print: ('Using http://production.example.com',)
............................................................F.............................................................
======================================================================
FAIL: test_invalid_permissions (test.test_sync.TestLocalFolder)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/miniconda/conda-bld/b2_1521727913797/test_tmp/test/test_sync.py", line 103, in test_invalid_permissions
    self.assertEqual(self.NAMES[1:], list(f.name for f in folder.all_files(self.reporter)))
AssertionError: Lists differ: ['hello.', 'hello/a/1', 'hello/a/2', 'hello/b', 'hello0', '自由'] != ['.dot_file', 'hello.', 'hello/a/1', 'hello/a/2', 'hello/b', 'hello0', '自由']

First differing element 0:
'hello.'
'.dot_file'

Second list contains 1 additional elements.
First extra element 6:
'自由'

- ['hello.', 'hello/a/1', 'hello/a/2', 'hello/b', 'hello0', '自由']
+ ['.dot_file', 'hello.', 'hello/a/1', 'hello/a/2', 'hello/b', 'hello0', '自由']
?  +++++++++++++


----------------------------------------------------------------------
Ran 201 tests in 1.037s

FAILED (failures=1)
```

Is this a known problem?

--- a/test/test_sync.py
+++ b/test/test_sync.py
@@ -107,10 +107,15 @@ class TestLocalFolder(TestSync):
     def test_invalid_permissions(self):
         with TempDir() as tmpdir:
             folder = self._prepare_folder(tmpdir, invalid_permissions=True)
-            self.assertEqual(self.NAMES[1:], list(f.name for f in folder.all_files(self.reporter)))
-            self.reporter.local_permission_error.assert_called_once_with(
-                os.path.join(tmpdir, self.NAMES[0])
-            )
+            if not os.access(os.path.join(tmpdir, self.NAMES[0]), os.R_OK):
+                self.assertEqual(
+                    self.NAMES[1:], list(f.name for f in folder.all_files(self.reporter))
+                )
+                self.reporter.local_permission_error.assert_called_once_with(
+                    os.path.join(tmpdir, self.NAMES[0])
+                )
+            else:
+                self.assertEqual(self.NAMES, list(f.name for f in folder.all_files(self.reporter)))
 
     def _check_file_filters_results(self, policies_manager, expected_scan_results):
         with TempDir() as tmpdir:

--- a/test/test_sync.py
+++ b/test/test_sync.py
@@ -107,6 +107,10 @@ class TestLocalFolder(TestSync):
     def test_invalid_permissions(self):
         with TempDir() as tmpdir:
             folder = self._prepare_folder(tmpdir, invalid_permissions=True)
+            # tests differ depending on the user running them. "root" will
+            # succeed in os.access(path, os.R_OK) even if the permissions of
+            # the file are 0 as implemented on self._prepare_folder().
+            # use-case: running test suite inside a docker container
             if not os.access(os.path.join(tmpdir, self.NAMES[0]), os.R_OK):
                 self.assertEqual(
                     self.NAMES[1:], list(f.name for f in folder.all_files(self.reporter))

